### PR TITLE
Fix: Auto-fill version details in rev.cpp and ottres.rc

### LINKED ISSUE #9062: "File version" is incorrect for Windows OpenTTD executable
## Version of OpenTTD
`OpenTTD 1.11.1` from the build artifacts for `openttd-1.11.1-windows-win64` or `openttd-1.11.1-windows-win32`

## Expected result
The "File version" that is part of the `openttd.exe` metadata should say `1.11.1.25330` (next: `1.11.2.25330`).

## Actual result
The "File version" that is part of the `openttd.exe` metadata says `1.11.0.25330`. Reinstalling the game from the installer still leaves the value `1.11.0.25330` for the file version, and so does installing the game from the zip archive.

## Steps to reproduce
1. Download `openttd-1.11.1-windows-win64` or `openttd-1.11.1-windows-win32` (zip archive or installer) onto a Windows 10 device.
2. Using Windows 10's File Explorer, look at the `File version` in the tooltip that comes up when you hover over `openttd.exe`.
3. Notice that it says `1.11.0.25330` and not the expected `1.11.1.25330` (next: `1.11.2.25330`).

### PR DESCRIPTION
**Pull request by glx22**
Fixes #9062.
Version for `release/1.11` branch [here](https://github.com/OpenTTD/OpenTTD/compare/release/1.11...glx22:autofill_version_1.11)
## Motivation / Problem
Version numbers are hard coded in many files, meaning we always forget to update some of them.

## Description
Auto detect and fill version numbers, there's still a hard coded default value but in only one location.

## Limitations
Only done `rev.cpp` and Windows specific file for now.

## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, gs_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

### CODE REVIEW

**Comment by LordAro** [comment]
Why not use the "builtin" variables - CPACK_PACKAGE_VERSION_MAJOR/MINOR/PATCH ? (Which then get combined into CPACK_PACKAGE_VERSION)

**Comment by glx22** [comment]
CPACK stuff is for the installer only, not the build itself, and cpack relies on whatever FindVersion.cmake detected. Also FindVersion.cmake generates the files before CPACK variables are set.

File: cmake/scripts/FindVersion.cmake [hunk 1]
```diff
@@ -124,6 +124,19 @@ if(GENERATE_OTTDREV)
     message(STATUS "Generating .ottdrev")
     file(WRITE ${CMAKE_SOURCE_DIR}/.ottdrev "${REV_VERSION}\t${REV_ISODATE}\t${REV_MODIFIED}\t${REV_HASH}\t${REV_ISTAG}\t${REV_ISSTABLETAG}\t${REV_YEAR}\n")
 else()
+    string(REGEX MATCH "^[0-9.]+" REV_SPLIT "${REV_VERSION}")
```
Comment (TrueBrain) [inline cmake/scripts/FindVersion.cmake]: Maybe use `REV_ISSTABLETAG` here to know if the version is expected to be a stable tag? No clue if that is easy to do :)

Comment (glx22) [reply cmake/scripts/FindVersion.cmake]: Could work, beta and RC would then default to hard coded version, which should not be an issue.

**Comment by nielsmh** [comment]
While touching this, I'd suggest maybe putting the branch name into the `SpecialBuild` version property when not building a tagged release version.

**Review by TrueBrain — APPROVED** [review]

**Pull request merged**